### PR TITLE
python 3 compatibility

### PR DESCRIPTION
**Pull request by dirk-thomas**

### CODE REVIEW

**Comment by tfoote** [comment]
+1

File: core/roslib/test/test_roslib_manifest.py [hunk 1]
```diff
@@ -143,8 +143,8 @@ def test_parse_bad_file(self):
       try:
         parse_file(p)
         self.fail("parse should have failed on bad manifest")
-      except ManifestException, e:
-        print str(e)
+      except ManifestException as e:
+        print(str(e))
```
Comment (wjwwood) [inline core/roslib/test/test_roslib_manifest.py:147]: Is it the case that `'{0}'.format(e)` would work better in Python2 if the exception contains unicode?

Comment (wjwwood) [inline core/roslib/test/test_roslib_manifest.py:147]: Actually I guess not unless you imported `unicode_literals` from `__future__`.

**Comment by wjwwood** [comment]
+1

**Comment by dirk-thomas** [comment]
Superseded by #45.

**Pull request closed without merging**